File: test/py/case10_not.py
# このファイルは `test/py/case10_not.py` のテスト/実装コードです。
# 役割が分かりやすいように、読み手向けの説明コメントを付与しています。
# 変更時は、既存仕様との整合性とテスト結果を必ず確認してください。

def invert(flag: bool) -> bool:
    if not flag:
        return True
    else:
        return False


if __name__ == "__main__":
    print(invert(False))

File: test/cpp/case10_not.cpp
#include "cpp_module/gc.h"
#include "cpp_module/py_runtime.h"
#include <algorithm>
#include <any>
#include <fstream>
#include <ios>
#include <iostream>
#include <sstream>
#include <stdexcept>
#include <string>
#include <tuple>
#include <type_traits>
#include <unordered_map>
#include <unordered_set>
#include <vector>

using namespace std;
using namespace pycs::gc;

bool invert(bool flag)
{
    if ((!flag))
    {
        return true;
    }
    else
    {
        return false;
    }
}

int main()
{
    py_print(invert(false));
    return 0;
}
